akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -0,0 +1,1577 @@
+     *
+     * If there is no current data value for the `key` the `initial` value will be
+     * passed to the `modify` function.
+     *
+     * The optional `request` context is included in the reply messages. This is a convenient
+     * way to pass contextual information (e.g. original sender) without having to use `ask`
+     * or local correlation data structures.
+     */
+    def this(key: String, readConsistency: ReadConsistency, writeConsistency: WriteConsistency,
+             request: Option[Any], modify: akka.japi.Function[Option[A], A]) =
+      this(key, readConsistency, writeConsistency, request)(data ⇒ modify.apply(data))
+
+    /**
+     * Java API: Modify value of local `Replicator` and replicate with given `writeConsistency`.
+     *
+     * If there is no current data value for the `key` the `initial` value will be
+     * passed to the `modify` function.
+     */
+    def this(
+      key: String, initial: A, writeConsistency: WriteConsistency, modify: akka.japi.Function[A, A]) =
+      this(key, ReadLocal, writeConsistency, None)(Update.modifyWithInitial(initial, data ⇒ modify.apply(data)))
+
+    /**
+     * Java API: Modify value of local `Replicator` and replicate with given `writeConsistency`.
+     *
+     * If there is no current data value for the `key` the `initial` value will be
+     * passed to the `modify` function.
+     *
+     * The optional `request` context is included in the reply messages. This is a convenient
+     * way to pass contextual information (e.g. original sender) without having to use `ask`
+     * or local correlation data structures.
+     */
+    def this(
+      key: String, initial: A, writeConsistency: WriteConsistency,
+      request: Option[Any], modify: akka.japi.Function[A, A]) =
+      this(key, ReadLocal, writeConsistency, request)(Update.modifyWithInitial(initial, data ⇒ modify.apply(data)))
+
+    /**
+     * Java API: Retrieve value from other replicas with given `readConsistency` and then modify value and
+     * replicate with given `writeConsistency`.
+     *
+     * If there is no current data value for the `key` the `initial` value will be
+     * passed to the `modify` function.
+     *
+     * The optional `request` context is included in the reply messages. This is a convenient
+     * way to pass contextual information (e.g. original sender) without having to use `ask`
+     * or local correlation data structures.
+     */
+    def this(
+      key: String, initial: A, readConsistency: ReadConsistency, writeConsistency: WriteConsistency,
+      request: Option[Any], modify: akka.japi.Function[A, A]) =
+      this(key, readConsistency, writeConsistency, request)(Update.modifyWithInitial(initial, data ⇒ modify.apply(data)))
+
+  }
+
+  sealed trait UpdateResponse extends NoSerializationVerificationNeeded {
+    def key: String
+  }
+  final case class UpdateSuccess(key: String, request: Option[Any]) extends UpdateResponse
```

Comment:
I don't see the use case where that should be needed, especially not when we remove the read part of `Update`. The current value will always be what the user changed it to in the `modify` function. It can be different on other nodes, but that is not something we should care about.